sim/src/plugin_wrapper.rs

```diff
@@ -141,11 +145,11 @@ mod tests {
     fn test_chained_plugins() {
         let mut cargo_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
         cargo_dir.push("../target/debug/libfilter_example");
-        let library_str = cargo_dir.to_str().unwrap();
-        let plugin1 = PluginWrapper::new(library_str, 0);
-        let plugin2 = PluginWrapper::new(library_str, 1);
-        let plugin3 = PluginWrapper::new(library_str, 2);
-        let plugin4 = PluginWrapper::new(library_str, 3);
+        let library_str = cargo_dir.to_str().unwrap().to_string();
+        let plugin1 = PluginWrapper::new("0".to_string(), library_str.clone());
```

Comment:
Same problem with excessive to_string() and clone(). Always a sign of code smell.